### ISSUE TITLE
windows - cmdline options seem not working correctly during installation

### ISSUE THREAD

**Issue by jirib** [grke/burp#696]
Hi,

Windows 2.1.30 client is not configured according to cmdline parameters during installation - bad 'server' and 'password' value.

I ran:

```
CMD /C "C:\Windows\Temp\burp-win64-installer-2.1.30.exe /S /server=172.16.10.2 /cname=w2k16-01 /password=xxxxxx"
```

And I got:

```
mode = client
server = 10.0.0.1
port = 4971
status_port = 4972
cname = w2k16-01
password = abcdefgh
...
```

**Comment by grke** [comment]
Hello,

Thanks for reporting. I will check when I have a chance.

Meanwhile, did you run the command over an existing installation?
The reason I ask is that it will not overwrite existing config, if I remember rightly.

**Comment by jirib** [comment]
Nope, clean install, I tried couple of times while 'Burp' was not present in 'Program Files' dir.

**Comment by deajan** [comment]
I can confirm the issue. Also affects the ```/nopowermode``` parameter I added, which doesn't work anymore. Issue appeared between 2.1.20 and 2.1.22.

**Comment by deajan** [comment]
Btw, I noticed that there isn't a docs/windows_installer.txt in the git repo, which I wanted to update because ```/nopowermode``` has never been documented.

**Comment by grke** [comment]
Most of the website bits are in the 'website' branch.
It might be confusing to look in there, because the whole website is based on a rubbish script that I wrote in 2001, or thereabouts. :)

**Comment by deajan** [comment]
I'll have a quick compile in order to check which commit bugs the command line switches :)

**Comment by deajan** [comment]
Removing this patch resolves the issue https://github.com/grke/burp/pull/613

**Comment by deajan** [comment]
Made a quick patch here #701

**Comment by grke** [comment]
Thanks for tracking that down. I don't mind doing a 2.1.32, but will wait until the end of the month in case more things show up.

**Comment by grke** [comment]
Deajan's patch is merged in 2.1-master, and the command line options.
However, I don't think the hostname thing is working.
I think it is supposed to fill in the cname field with the hostname, but I always get 'clientname' if I do not specify it on the command line.

**Comment by ziirish** [comment]
I don't know much about Windows but my understanding of the installation script thing is:

1. if you don't specify a `/cname` option then the value 'clientname' is used
2. if you specify a `/cname=toto` option, then the value 'toto' is used
3. if you specify `/cname` option (without any value), then the hostname of the current machine is used

Am I right?
Is this what you tried Graham?

**Comment by deajan** [comment]
I have compiled the patch for v2.1.28 and 2.1.30 and tested it on Windows 10 x64.
Running the installer straight from GUI without any options fills my hostname (case sensitive) into cname field. Using /cname overrides my hostname with whatever I specified.

What system have you tried it on ?

**Comment by jirib** [comment]
I"m not expert in CI but https://github.com/marketplace/appveyor seems to offer to OSS free CI, thus Windows build could be tested there as Linux/OS X one via Travis.

**Comment by grke** [comment]
I have my own CI for Windows already. It's just a case of adding new tests to it if I have time.

**Comment by grke** [comment]
I tried the installer on Windows 7, without using any command line switches so that you get the windows clicky box gui things - I get 'clientname' in the field (and hence the burp.conf).

I tried from the command line, setting /cname=something, I get 'something' in the burp.conf.
I tried from the command line, not setting /cname, I get 'clientname' in the burp.conf.

I will try again, in case I was making some mistake.

Maybe the hostname thing doesn't work on Windows 7 because it is old?
I will try Windows 10 as well.

Ziirish, I don't expect to see 'clientname' unless the hostname thing is unable to get a result for some reason (such as the Windows machine being too old, I guess).
